cxx/src/lib/utility/MetadataDefinitions.cc

```diff
@@ -1,26 +1,22 @@
+#include <string.h>
 #include <fstream>
 #include <sstream>
+#include "mspass/utility/utility.h"
 #include "mspass/utility/MsPASSError.h"
 #include "mspass/utility/AntelopePf.h"
+#include "yaml-cpp/yaml.h"
 #include "mspass/utility/MetadataDefinitions.h"
 namespace mspass{
 using namespace mspass;
-using std::string;
-/* This is a temporary implementation to test a theory for why this created a
-seg fault*/
-MetadataDefinitions::MetadataDefinitions()
-{
-  /*
-  MetadataDefinitions temp(string("obspy_namespace.pf"),MDDefFormat::PF);
-  *this=temp;
-  */
-}
+using namespace std;
 MetadataDefinitions::MetadataDefinitions(const std::string mdname)
 {
     try{
-        /* This s temporary.  For now this defaults to pf format file input */
-        MetadataDefinitions tmp(mdname,MDDefFormat::PF);
-        *this=tmp;
+      string datadir=mspass::data_directory();
+      string path;
+      path=datadir+"/yaml/"+mdname+".yaml";
```

Comment:
if a user explicitly give the full path in the input `mdname`, isn't it gonna break?